### PR TITLE
Single Step Instrumentation Compatibility

### PR DESCRIPTION
**Pull request by brett0000FF**
### What does this PR do? What is the motivation?

Create new page with more detailed information on Single Step Instrumentation compatibility.

### Merge instructions

Merge queue is enabled in this repo. To have it automatically merged after it receives the required reviews, create the PR (from a branch that follows the `<yourname>/description` naming convention) and then add the following PR comment:

```
/merge
```

### CODE REVIEW

File: content/en/tracing/trace_collection/automatic_instrumentation/single-step-apm/compatibility.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+{{< /programming-lang >}}
+
+{{< programming-lang lang="dotnet" >}}
+
+### Supported .NET runtimes
+
+| .NET Version     | Support                         | Notes                               |
+|------------------|---------------------------------|-------------------------------------|
+| .NET 8           | <i class="icon-check-bold"></i> |                                     |
```
Comment (andrewlock) [inline content/en/tracing/trace_collection/automatic_instrumentation/single-step-apm/compatibility.md:73]: Just an FYI, .NET 9 is released next week, and we're _hoping_ to add support for it ASAP afterwards, i.e. pre SSI GA

File: content/en/tracing/trace_collection/automatic_instrumentation/single-step-apm/compatibility.md [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+### Supported .NET runtimes
+
+| .NET Version     | Support                         | Notes                                |
+|------------------|---------------------------------|--------------------------------------|
+| .NET 8           | <i class="icon-check-bold"></i> |                                      |
+| .NET 7           | <i class="icon-check-bold"></i> |                                      |
+| .NET 6           | <i class="icon-check-bold"></i> | Versions below 6.0.13 not supported. |
+| .NET 5           | <i class="icon-check-bold"></i> |                                      |
+| .NET Core 3.1    | <i class="icon-check-bold"></i> |                                      |
+| .NET Core 3.0    |                                 |                                      |
```
Comment (janine-c) [inline content/en/tracing/trace_collection/automatic_instrumentation/single-step-apm/compatibility.md:148]: I'm not sure how to interpret rows that neither have the check mark nor "not supported" - which are they? Are they in a weird third state?

**Review by janine-c — CHANGES_REQUESTED** [review]
Hey Brett, this looks great! I made some comments about some things I didn't understand, but feel free to ignore me if I was nitpicking things that are standard in this kind of documentation :)

File: config/_default/menus/main.en.yaml [hunk 3]
```diff
@@ -3176,7 +3176,7 @@ menu:
       identifier: automatic_instrumentation
       weight: 201
     - name: Using Single Step Instrumentation
-      url: tracing/trace_collection/automatic_instrumentation/single-step-apm/
+      url: /tracing/trace_collection/automatic_instrumentation/single-step-apm/
```
[the remark this answers is not in the document]
Comment (brett0000FF) [reply config/_default/menus/main.en.yaml:3179]: Noted! Good catch. It should work either way, but it's best to be consistent.

File: content/en/tracing/trace_collection/automatic_instrumentation/single-step-apm/compatibility.md [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+
+This section provides language-specific compatibility requirements for Single Step Instrumentation:
+
+{{< programming-lang-wrapper langs="java,python,ruby,nodejs,dotnet,php" >}}
+
+{{< programming-lang lang="java" >}}
+
+### Supported Java versions
+
+| Java Version | Support                         | Notes |
```
[the remark this answers is not in the document]
Comment (brett0000FF) [reply content/en/tracing/trace_collection/automatic_instrumentation/single-step-apm/compatibility.md]: Removed for now.

File: content/en/tracing/trace_collection/automatic_instrumentation/single-step-apm/compatibility.md [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+| 8+           | <i class="icon-check-bold"></i> |       |
+
+### Troubleshooting
+
+**Environment Variable Length**: If your application uses extensive command-line options or environment variables, you might encounter initialization failures. This typically occurs when you have many JVM arguments or other startup configurations. To resolve this:
+  - Review and minimize non-essential JVM arguments
+  - Consider moving some configurations to a properties file
+  - Check application logs for specific initialization errors
+
+<div class="alert alert-info">Single Step Instrumentation disables automatically when custom instrumentation is detected.</div>
```
[the remark this answers is not in the document]
Comment (brett0000FF) [reply content/en/tracing/trace_collection/automatic_instrumentation/single-step-apm/compatibility.md]: Let me know what you think about the new phrasing. Agreed it was super passive before.

File: content/en/tracing/trace_collection/automatic_instrumentation/single-step-apm/compatibility.md (hunk 2 above)
Comment (brett0000FF) [reply content/en/tracing/trace_collection/automatic_instrumentation/single-step-apm/compatibility.md:148]: Great note. I don't even think we need to mention that the preview versions are explicitly unsupported. I removed that row, so it should be more clear now.

**Review by janine-c — APPROVED** [review]
Looks awesome! :shipit:

**Pull request merged**